From 121b4832ba46ba853d1f5f76292375b8ffedb2b6 Mon Sep 17 00:00:00 2001
From: Tony Breeds <tony@bakeyournoodle.com>
Date: Tue, 30 Sep 2008 17:00:19 -0400
Subject: [PATCH] From tony@bakeyournoodle.com Wed Sep 26 10:26:59 2007
 Date: Tue, 04 Sep 2007 17:09:02 +1000
 To: linux-rt-users@vger.kernel.org
 Subject: [PATCH 2/5] count_active_rt_tasks() is undefined when
     CONFIG_PREEMPT_RT is not set.
 Also, it looks to me that active_rt_tasks[] was never modified.

Signed-off-by: Tony Breeds <tony@bakeyournoodle.com>
---
 kernel/timer.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/kernel/timer.c b/kernel/timer.c
index d3f4dac..2f896db 100644
--- a/kernel/timer.c
+++ b/kernel/timer.c
@@ -1099,21 +1099,25 @@ unsigned long avenrun_rt[3];
 static inline void calc_load(unsigned long ticks)
 {
 	unsigned long active_tasks; /* fixed-point */
-	unsigned long active_rt_tasks; /* fixed-point */
 	static int count = LOAD_FREQ;
+#ifdef CONFIG_PREEMPT_RT
+	unsigned long active_rt_tasks; /* fixed-point */
+#endif
 
 	count -= ticks;
 	if (unlikely(count < 0)) {
 		active_tasks = count_active_tasks();
+#ifdef CONFIG_PREEMPT_RT
 		active_rt_tasks = count_active_rt_tasks();
+#endif
 		do {
 			CALC_LOAD(avenrun[0], EXP_1, active_tasks);
 			CALC_LOAD(avenrun[1], EXP_5, active_tasks);
 			CALC_LOAD(avenrun[2], EXP_15, active_tasks);
 #ifdef CONFIG_PREEMPT_RT
-			CALC_LOAD(avenrun_rt[0], EXP_1, active_tasks);
-			CALC_LOAD(avenrun_rt[1], EXP_5, active_tasks);
-			CALC_LOAD(avenrun_rt[2], EXP_15, active_tasks);
+			CALC_LOAD(avenrun_rt[0], EXP_1, active_rt_tasks);
+			CALC_LOAD(avenrun_rt[1], EXP_5, active_rt_tasks);
+			CALC_LOAD(avenrun_rt[2], EXP_15, active_rt_tasks);
 #endif
 			count += LOAD_FREQ;
 
-- 
1.6.0.90.g436ed

